u2ftoken/example/main.go

```diff
@@ -75,17 +72,23 @@ func main() {
 	}
 	t = u2ftoken.NewToken(dev)
 
-	io.ReadFull(rand.Reader, challenge)
+	if _, err := io.ReadFull(rand.Reader, challenge); err != nil {
+		log.Fatal(err)
+	}
+
 	req := u2ftoken.AuthenticateRequest{
 		Challenge:   challenge,
 		Application: app,
-		KeyHandle:   keyHandle,
+		KeyHandle:   res.KeyHandle,
 	}
 	if err := t.CheckAuthenticate(req); err != nil {
 		log.Fatal(err)
 	}
 
-	io.ReadFull(rand.Reader, challenge)
+	if _, err := io.ReadFull(rand.Reader, challenge); err != nil {
+		log.Fatal(err)
+	}
+
```

Comment:
nit: no need to generate a new challenge here, it's already populated above (line 75)